base/nullable.jl

```diff
@@ -63,16 +63,79 @@ get(x::Nullable) = x.isnull ? throw(NullException()) : x.value
 
 isnull(x::Nullable) = x.isnull
 
-function isequal(x::Nullable, y::Nullable)
-    if x.isnull && y.isnull
-        return true
-    elseif x.isnull || y.isnull
-        return false
+
+## Operators
+
+"""
+    null_safe_op(f::Any, ::Type)::Bool
+    null_safe_op(f::Any, ::Type, ::Type)::Bool
+
+Returns whether an operation `f` can safely be applied to any value of the passed type(s).
+Returns `false` by default.
+
+Custom types should implement methods for some or all operations `f` when applicable:
+returning `true` means that the operation may be called on any bit pattern without
+throwing an error (though returning invalid or nonsensical results is not a problem).
+In particular, this means that the operation can be applied on the whole domain of the
+type *and on uninitialized objects*. As a general rule, these proporties are only true for
+safe operations on `isbits` types.
+
+Types declared as safe can benefit from higher performance for operations on nullable: by
+always computing the result even for null values, a branch is avoided, which helps
+vectorization.
+"""
+null_safe_op(f::Any, ::Type) = false
+null_safe_op(f::Any, ::Type, ::Type) = false
+
+typealias NullSafeSignedInts Union{Int128, Int16, Int32, Int64, Int8}
+typealias NullSafeUnsignedInts Union{Bool, UInt128, UInt16, UInt32, UInt64, UInt8}
+typealias NullSafeInts Union{NullSafeSignedInts, NullSafeUnsignedInts}
+typealias NullSafeFloats Union{Float16, Float32, Float64}
+typealias NullSafeTypes Union{NullSafeInts, NullSafeFloats}
+
+null_safe_op{S<:NullSafeTypes,
+             T<:NullSafeTypes}(::typeof(isequal), ::Type{S}, ::Type{T}) = true
+
+"""
+    isequal(x::Nullable, y::Nullable)
+
+If neither `x` nor `y` is null, compares them according to their values
```

Comment:
I think the preferred way to describe functionality in a docstring is to use imperative rather than declarative sentences, e.g. "If neither `x` nor `y` is null, **compare** them," versus "If neither `x` nor `y` is null, **compares** them."